src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,990 @@ const ActorInit Dm_Char08_InitVars = {
+    }
+    DmChar08_SetAnimation(&this->skelAnime, &sLargeTurtleAnimationInfo[this->animIndex], 0);
+}
+
+void DmChar08_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DmChar08* this = THIS;
+
+    if (this->dynapolyInitialized) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+s16 D_80AB1770 = 0; // Some kind of latch for playing a sound only once?
+
+void func_80AAF610(DmChar08* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Player* player2 = GET_PLAYER(globalCtx);
+
+    if ((player2->stateFlags2 & 0x8000000) &&
+        ((player2->actor.world.pos.x > -5780.0f) && (player2->actor.world.pos.x < -5385.0f) &&
+         (player2->actor.world.pos.z > 1120.0f) && (player2->actor.world.pos.z < 2100.0f))) {
+        if (D_80AB1770 == 0) {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            D_80AB1770 = 1;
+        }
+    } else {
+        D_80AB1770 = 0;
+    }
+    if ((player->transformation == PLAYER_FORM_ZORA) && (globalCtx->msgCtx.ocarinaMode == 3) &&
+        (globalCtx->msgCtx.unk1202E == 2)) {
+        if ((player2->actor.world.pos.x > -5780.0f) && (player2->actor.world.pos.x < -5385.0f)) {
+            if ((player2->actor.world.pos.z > 1120.0f) && (player2->actor.world.pos.z < 2100.0f)) {
+                this->actionFunc = func_80AAF79C;
+            }
+        }
+    }
+}
+
+void func_80AAF79C(DmChar08* this, GlobalContext* globalCtx) {
+    s16 cs1 = this->dyna.actor.cutscene;
+    s16 cs = ActorCutscene_GetAdditionalCutscene(
+        ActorCutscene_GetAdditionalCutscene(ActorCutscene_GetAdditionalCutscene(cs1)));
+
+    if (gSaveContext.save.weekEventReg[93] & 8) {
+        cs1 = cs;
+    }
+
+    if (ActorCutscene_GetCanPlayNext(cs1)) {
+        ActorCutscene_Start(cs1, &this->dyna.actor);
+        gSaveContext.save.weekEventReg[53] |= 0x20;
+        gSaveContext.save.weekEventReg[93] |= 8;
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->actionFunc = func_80AAF884;
+    } else {
+        ActorCutscene_SetIntentToPlay(cs1);
+    }
+}
 
-extern InitChainEntry D_80AB1764[];
+void func_80AAF884(DmChar08* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        DynaPolyActor_Init(&this->dyna, 3);
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_kamejima_Colheader_002470);
+        this->dyna.actor.flags |= ACTOR_FLAG_1;
+        this->actionFunc = func_80AAF8F4;
+    }
+}
 
-extern UNK_TYPE D_06002470;
-extern UNK_TYPE D_06004E70;
-extern UNK_TYPE D_0600E748;
+void func_80AAF8F4(DmChar08* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 yOffset;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF050.s")
+    if (fabsf(this->dyna.actor.xzDistToPlayer) < 200.0f) {
+        this->actionFunc = func_80AAFA18;
+    }
+    this->unk_1FA += 0xC8;
+    yOffset = 50.0f;
+    yOffset *= Math_SinS(this->unk_1FA);
```

Comment:
Looks like this is just some sort of bob timer/phase. It's not used anywhere else so I'd just put it in dec and (also nice assumption of the struct being initialised to 0...)